src/master-playlist-controller.js

```diff
@@ -794,23 +794,20 @@ export class MasterPlaylistController extends videojs.EventTarget {
       return this.mediaSource.endOfStream('network');
     }
 
-    // Blacklist this playlist
-    currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
+    if (!this.masterPlaylistLoader_.isFinalRendition_()) {
+      // Blacklist this playlist
+      currentPlaylist.excludeUntil = Date.now() + BLACKLIST_DURATION;
 
-    // Select a new playlist
-    nextPlaylist = this.selectPlaylist();
+      // Select a new playlist
+      nextPlaylist = this.selectPlaylist();
 
-    if (nextPlaylist) {
       videojs.log.warn('Problem encountered with the current ' +
                        'HLS playlist. Switching to another playlist.');
 
       return this.masterPlaylistLoader_.media(nextPlaylist);
     }
-    videojs.log.warn('Problem encountered with the current ' +
-                     'HLS playlist. No suitable alternatives found.');
-    // We have no more playlists we can select so we must fail
-    this.error = error;
-    return this.mediaSource.endOfStream('network');
+    // Never blacklisting this playlist because it's final rendition
+    return this.masterPlaylistLoader_.media(currentPlaylist);
```

Comment:
This definitely has the effect we are going for, but changing the playlist loader's state outside of itself like this can make debugging difficult because now you have to know about this one line of code in a separate file that changes the state. I think we can safely restore the state in certain error conditions without introducing regressions. There are only a few places that errors trigger in `PlaylistLoader` and some places already use the `playlistRequestError` function which the optional third parameter takes a state value to reset that to.  I think you should be able to safely reset the state in each error condition using this.